deepchem/models/torch_models/layers.py

```diff
@@ -2796,3 +2796,173 @@ def forward(
             else:
                 out_tensor += w * in_tensor
         return out_tensor
+
+
+class SetGather(nn.Module):
+    """set2set gather layer for graph-based model
+
+    Models using this layer must set `pad_batches=True`
+
+    Torch Equivalent of Keras SetGather layer
+
+    Examples
+    --------
+    >>> import deepchem as dc
+    >>> import numpy as np
+    >>> from deepchem.models.torch_models import layers
+    >>> total_n_atoms = 4
+    >>> n_atom_feat = 4
+    >>> atom_feat = np.random.rand(total_n_atoms, n_atom_feat)
+    >>> atom_split = np.array([0, 0, 1, 1], dtype=np.int32)
+    >>> gather = layers.SetGather(2, 2, n_hidden=4)
+    >>> output_molecules = gather([atom_feat, atom_split])
+    >>> print(output_molecules.shape)
+    torch.Size([2, 8])
+
+    """
+
+    def __init__(self,
+                 M: int,
+                 batch_size: int,
+                 n_hidden: int = 100,
+                 init='orthogonal',
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        M: int
+            Number of LSTM steps
+        batch_size: int
+            Number of samples in a batch(all batches must have same size)
+        n_hidden: int, optional
+            number of hidden units in the passing phase
+
+        """
+
+        super(SetGather, self).__init__(**kwargs)
+        self.M = M
+        self.batch_size = batch_size
+        self.n_hidden = n_hidden
+        self.init = init
+
+        def initFunc(input_shape):
+            return nn.Parameter(
+                torch.Tensor(input_shape[0], input_shape[1]).normal_(mean=0.0,
+                                                                     std=0.1))
+
+        self.U = initFunc((2 * self.n_hidden, 4 * self.n_hidden))
```

Comment:
simply set self.U as nn.Parameter(...), seems there's no need to define a separate function for this.

Reply:
Done.